### DISCUSSION TITLE
Field BT-02-notice and BT-03-notice

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#349] [Q&A] [answered]
Hi labs !
I don't understand why those fields have got "displayType" : "COMBOBOX".
Once a notice is chosen do they not fixed ? 
for example , subtype 1 : 
  "noticeSubTypes" : [ {
    "documentType" : "PIN",
    "legalBasis" : "32014L0024",
    **"formType" : "planning",
    "type" : "pin-buyer",**
    "description" : "Notice of the publication of a prior information notice on a buyer profile - general directive",
    "subTypeId" : "1",
    "_label" : "notice|name|1",
    "viewTemplateIds" : [ "1", "summary" ]
  }

Thanks and regards
Renato

**Comment by rouschr** [comment, marked as answer]
This is because normally a field of type "code" is associated with a combobox to select a code from a codelist:
```
 "id" : "BT-02-notice",
    "parentNodeId" : "ND-Root",
    "name" : "Notice Type",
    "btId" : "BT-02",
    "xpathAbsolute" : "/*/cbc:NoticeTypeCode",
    "xpathRelative" : "cbc:NoticeTypeCode",
    "type" : "code",
    "legalType" : "CODE",
    "repeatable" : {
      "value" : false,
      "severity" : "ERROR"
    },...
```
In the notice-types **metadata**:
```
 "metadata" : [ {
    "id" : "BT-02-notice",
    "contentType" : "field",
    "displayType" : "COMBOBOX",
    "description" : "Notice Type",
    "_label" : "field|name|BT-02-notice",
    "readOnly" : true <---------------------------
  }
```
So this is to be set by your UI system and is not modifiable by users because it is **readOnly**.
I think here COMBOBOX is just to ensure that the selected code is really part of the list of available codes in the combobox (as in general UI systems do this). It would probably also work with a simple text box.
